packer-scripts/minimize

```diff
@@ -1,38 +1,47 @@
 #!/usr/bin/env bash
-
 set -o errexit
-set -o xtrace
 
-if [[ $PACKER_BUILDER_TYPE = googlecompute || $PACKER_BUILDER_TYPE = amazon-ebs ]]; then
-  echo "Skipping minimization on $PACKER_BUILDER_TYPE"
-  exit 0
-fi
+main() {
+  set -o xtrace
 
-if [[ $PACKER_BUILDER_TYPE != docker ]]; then
-  readonly SWAP_UUID=$(/sbin/blkid -o value -l -s UUID -t TYPE=swap)
+  if [[ "x${PACKER_BUILDER_TYPE}" = xgooglecompute || "x${PACKER_BUILDER_TYPE}" = xamazon-ebs ]]; then
+    echo "Skipping minimization on ${PACKER_BUILDER_TYPE}"
+    exit 0
+  fi
 
-  if [[ $SWAP_UUID ]]; then
-    readonly SWAP_PART=$(readlink -f /dev/disk/by-uuid/"$SWAP_UUID")
-    /sbin/swapoff "$SWAP_PART"
-    dd if=/dev/zero of="$SWAP_PART" bs=1M || echo "dd exit $? suppressed"
-    /sbin/mkswap -U "$SWAP_UUID" "$SWAP_PART"
+  if [[ "x${PACKER_BUILDER_TYPE}" != xdocker ]]; then
+    readonly SWAP_UUID
+    SWAP_UUID="$(/sbin/blkid -o value -l -s UUID -t TYPE=swap)"
+
+    if [[ "${SWAP_UUID}" ]]; then
+      readonly SWAP_PART
+      SWAP_PART="$(readlink -f "/dev/disk/by-uuid/${SWAP_UUID}")"
+      /sbin/swapoff "${SWAP_PART}"
+      dd if=/dev/zero of="${SWAP_PART}" bs=1M \
+        || echo "dd exit $? suppressed"
+      /sbin/mkswap -U "${SWAP_UUID}" "${SWAP_PART}"
+    fi
   fi
-fi
 
-dd if=/dev/zero of=/EMPTY bs=1M &
-export DD_PID="$!"
-echo 'snoozing a bit here for dd mkay'
-sleep 5
+  dd if=/dev/zero of=/EMPTY bs=1M &
+  export DD_PID="$!"
+  echo 'snoozing a bit here for dd mkay'
+  sleep 5
 
-trap 'kill -9 '"${DD_PID}"' &>/dev/null || true' EXIT QUIT TERM INT
+  trap 'kill -9 '"${DD_PID}"' &>/dev/null || true' EXIT QUIT TERM INT
 
-while pgrep dd; do
-  kill -USR1 "$DD_PID" || true
-  sleep 5
-done
+  while pgrep dd; do
```

Comment:
This could potentially break if other `dd` operations are running at the same time (seems unlikely, but I don't know if a cron job could happen at the same time or something? `while kill -0 "${DD_PID}"` might be better? (`kill -0 $PID` will exit with 0 if a signal could be sent, but doesn't actually send a signal, so it's an easy way to see if a process exists and you are allowed to send signals to it).